tests/php/Security/PasswordExpirationMiddlewareTest.php

```diff
@@ -0,0 +1,517 @@
+<?php
+
+namespace SilverStripe\Security\Tests;
+
+use SilverStripe\Control\Director;
+use SilverStripe\Control\Tests\HttpRequestMockBuilder;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\ValidationException;
+use SilverStripe\Security\PasswordExpirationMiddleware;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+
+class PasswordExpirationMiddlewareTest extends SapphireTest
+{
+    use HttpRequestMockBuilder;
+
+    private $rollbackMember;
+
+    private $rollbackEnable;
+
+    private $rollbackDefaultRedirect;
+
+    private $rollbackWhitelistedUrls;
+
+    private $rollbackWhitelistedUrlStartswith;
+
+    protected function setUp()
+    {
+        Director::config()->set('alternate_base_url', 'http://localhost/custom-base/');
+        $this->rollbackMember = Security::getCurrentUser();
+        $this->rollbackEnable =
+            PasswordExpirationMiddleware::config()->get('enable');
+        $this->rollbackDefaultRedirect =
+            PasswordExpirationMiddleware::config()->get('default_redirect');
+        $this->rollbackWhitelistedUrls =
+            PasswordExpirationMiddleware::config()->get('whitelisted_urls');
+        $this->rollbackWhitelistedUrlStartswith =
+            PasswordExpirationMiddleware::config()->get('whitelisted_url_startswith');
+
+        PasswordExpirationMiddleware::config()->set('enable', true);
+        PasswordExpirationMiddleware::config()->set('default_redirect', null);
+        PasswordExpirationMiddleware::config()->set('whitelisted_urls', []);
+        PasswordExpirationMiddleware::config()->set('whitelisted_url_startswith', []);
+    }
+
+    protected function tearDown()
+    {
+        Director::config()->remove('alternate_base_url');
+        Security::setCurrentUser($this->rollbackMember);
+        PasswordExpirationMiddleware::config()->set(
+            'enable',
+            $this->rollbackEnable
+        );
+        PasswordExpirationMiddleware::config()->set(
+            'default_redirect',
+            $this->rollbackDefaultRedirect
+        );
+        PasswordExpirationMiddleware::config()->set(
+            'whitelisted_urls',
+            $this->rollbackWhitelistedUrls
+        );
+        PasswordExpirationMiddleware::config()->set(
+            'whitelisted_url_startswith',
+            $this->rollbackWhitelistedUrlStartswith
+        );
```

Comment:
When you call `parent::setUp()` the config layer is automatically nested, so you don't need to reset any configuration values afterwards (including the current user)